zulipterminal/helper.py

```diff
@@ -732,3 +734,52 @@ def suppress_output() -> Iterator[None]:
     finally:
         os.dup2(stdout, 1)
         os.dup2(stderr, 2)
+
+
+@asynch
+def process_media(controller: Any, media_link: str) -> None:
+    media_name = media_link.split("/")[-1]
+    client = controller.client
+    auth = requests.auth.HTTPBasicAuth(client.email, client.api_key)
+
+    with requests.get(media_link, auth=auth, stream=True) as r:
+        r.raise_for_status()
+        with NamedTemporaryFile(
+            mode="wb", delete=False, prefix="zt-", suffix="-{}".format(media_name)
+        ) as f:
+            media_path = f.name
+            for chunk in r.iter_content(chunk_size=8192):
+                if chunk:  # Filter out keep-alive new chunks.
+                    f.write(chunk)
+
+    if WSL:
+        command = f'explorer.exe `wslpath -w "{media_path}"`'  # Needs testing.
+    elif LINUX:
+        command = f"xdg-open {media_path}"
+    elif MACOS:
+        command = f"open {media_path}"
+
+    tool = command.split()[0]
+    open_media(controller, command, tool)
+
+
+@asynch
+def open_media(controller: Any, command: str, tool: str) -> None:
```

Comment:
This is open_media, so it's surprising that the open-command selection is not present here.

It's been a while since I've looked at this code, but it may be clearer not having one function call the other, ie. download_media(url) -> local_path, open_media(local_path) ?